packages/grafana-runtime/src/services/LocationService.tsx

```diff
@@ -153,10 +154,37 @@ export function locationSearchToObject(search: string | number): UrlQueryMap {
   return {};
 }
 
+const history = H.createBrowserHistory({ basename: config.appSubUrl ?? '/' });
+
+// Setup for having 2 sub URLs at the same time in a single main URL. This is used for the app sidecar feature and so
+// won't work without a feature flag. Without the feature flag there should be no functional change.
+
+// Name of the query param under which the secondary URL is stored. Should be something that is unlikely to be used
+// for normally.
+const secondaryUrlParam = '__sc';
+const mainHistory = config.featureToggles.appSidecar
+  ? createAggregateHistory({
+      actualHistory: history,
+      isMain: true,
+      param: secondaryUrlParam,
+    })
+  : history;
+const secondaryHistory = createAggregateHistory({
+  actualHistory: history,
+  isMain: false,
+  param: secondaryUrlParam,
+});
```

Comment:
I can't see these global vars being accessed anywhere. 

I know we have these global service singletons already (I wish we did not), should try to limit creating new ones and esp create new objects in global scope (if we can).